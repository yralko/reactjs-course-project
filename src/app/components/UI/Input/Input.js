import React from 'react';
import classes from './index.css';

const Input = props => {
  const styles = [classes.Input];

  if(props.styles) styles.push(props.styles);

  return (
    <input
      className={styles.join(' ')}
      onChange={(e) => props.changed(e)}
      onKeyUp={(e) => props.keyReleased(e)}
      type={props.type}
      placeholder={props.placeholder}
    />
  );
}

export default Input;
