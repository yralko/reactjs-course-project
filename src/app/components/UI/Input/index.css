.Input {
  border: none;
  outline: none;
}
