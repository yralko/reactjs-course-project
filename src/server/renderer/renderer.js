import React from 'react';
import { renderToString } from 'react-dom/server';
import { Provider } from 'react-redux';
import { StaticRouter as Router } from 'react-router-dom';
import App from '../../app/App';


export default (req, store, context) => {
  const root = (
    <Provider store={store}>
      <Router location={req.url} context={context}>
        <App />
      </Router>
    </Provider>
  );

  const content = renderToString(root); 

  return `
  <!doctype html>
  <html lang="en">
  <head>
    <meta charset="utf-8">
    <title>The HTML5 Herald</title>
    <base href="/" />
    <link rel="stylesheet" href="style.css" />
  </head>
  <body>
    <div id="root">${content}</div>
    <script>window.INITIAL_STATE = ${JSON.stringify(store.getState())}</script>
    <script src="bundle.js"></script>
  </body>
  </html>
  `;
};
