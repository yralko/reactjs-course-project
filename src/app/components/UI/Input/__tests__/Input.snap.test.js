import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../Input';

test('matches snapshot', () => {
  const tree = renderer.create(<Input />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
