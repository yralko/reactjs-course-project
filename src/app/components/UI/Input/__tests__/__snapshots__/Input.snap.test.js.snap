// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<input
  className={undefined}
  onChange={[Function]}
  onKeyUp={[Function]}
  placeholder={undefined}
  type={undefined}
/>
`;
